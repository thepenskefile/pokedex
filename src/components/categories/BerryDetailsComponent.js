// @flow

import React from 'react';
import { Container, Box, Image, Flex, Heading, Set, Text, Columns, Column, Tag, palette } from 'fannypack';
import Counter from '../../ui/Counter';

type Props = {
  data: Object
};

const MAX_BERRY_SIZE = 250;
const MAX_BERRY_SMOOTHNESS = 85;

const BerryDetailsComponent = ({ data }: Props) => {
  return (
    <Container textAlign="left" marginLeft="10px" width="96%">
      <Flex alignItems="center">
        <Heading>{data.name.charAt(0).toUpperCase() + data.name.slice(1)}</Heading>
        <Image src={`https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/items/${data.item.name}.png`} />
      </Flex>
      <Set marginBottom="20px" spacing="major-2" display="block">
        <Text use="strong">Growth time: </Text>
        {data.growth_time}
        <Text use="strong">Max harvest: </Text>
        {data.max_harvest}
      </Set>
      <Set marginBottom="20px">
        {data.flavors.map(
          (flavor, i) =>
            flavor.potency > 0 && (
              <Tag key={flavor.flavor.name} backgroundColor={flavor.flavor.name} size="medium">
                {flavor.flavor.name}
              </Tag>
            )
        )}
      </Set>
      <Box marginTop="40px">
        <Columns marginTop="20px" marginBottom="20px" minBreakpoint="mobile">
          <Column display="inline-block" spread={2} spreadTablet={4} spreadMobile={6} spreadWidescreen={3}>
            <Counter
              percentage={Math.round((data.size / MAX_BERRY_SIZE) * 100)}
              label={'size'}
              color={data.flavors[0].flavor.name ? palette(`${data.flavors[0].flavor.name}`) : palette('normal')}
            />
          </Column>
          <Column display="inline-block" spread={2} spreadTablet={4} spreadMobile={6} spreadWidescreen={3}>
            <Counter
              percentage={Math.round((data.smoothness / MAX_BERRY_SMOOTHNESS) * 100)}
              label={'smoothness'}
              color={data.flavors[0].flavor.name ? palette(`${data.flavors[0].flavor.name}`) : palette('normal')}
            />
          </Column>
          <Column display="inline-block" spread={2} spreadTablet={4} spreadMobile={6} spreadWidescreen={3}>
            <Counter
              percentage={data.soil_dryness}
              label={'soil dryness'}
              color={data.flavors[0].flavor.name ? palette(`${data.flavors[0].flavor.name}`) : palette('normal')}
            />
          </Column>
        </Columns>
      </Box>
    </Container>
  );
};

export default BerryDetailsComponent;
