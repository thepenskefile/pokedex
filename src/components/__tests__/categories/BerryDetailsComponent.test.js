import React from 'react';
import BerryDetailsComponent from '../../categories/BerryDetailsComponent';
import renderer from 'react-test-renderer';

const defaultProps = {
  data: {
    name: 'mockName',
    growth_time: 'mockGrowthTime',
    max_harvest: 'mockMaxHarvest',
    size: 'mockSize',
    smoothness: 'mockSmoothness',
    soil_dryness: 'mockSoilDryness'
  }
};

it('renders correctly', () => {
  const component = renderer.create(<BerryDetailsComponent {...defaultProps} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
