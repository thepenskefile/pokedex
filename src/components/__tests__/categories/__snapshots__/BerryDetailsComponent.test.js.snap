// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="styled-sc-144wh3b-0 iUlbjp Box-sc-6sssmg-0 inpIDs"
  style={
    Object {
      "marginLeft": "10px",
      "textAlign": "left",
      "width": "96%",
    }
  }
  width="96%"
>
  <div
    className="Box-sc-6sssmg-0 inpIDs"
  >
    Name: 
    mockName
  </div>
  <div
    className="Box-sc-6sssmg-0 inpIDs"
  >
    Growth Time: 
    mockGrowthTime
  </div>
  <div
    className="Box-sc-6sssmg-0 inpIDs"
  >
    Max Harvest: 
    mockMaxHarvest
  </div>
  <div
    className="Box-sc-6sssmg-0 inpIDs"
  >
    Size: 
    mockSize
  </div>
  <div
    className="Box-sc-6sssmg-0 inpIDs"
  >
    Smoothness: 
    mockSmoothness
  </div>
  <div
    className="Box-sc-6sssmg-0 inpIDs"
  >
    Soil dryness: 
    mockSoilDryness
  </div>
</div>
`;
